Framework/PythonInterface/plugins/algorithms/FindGlobalBMatrix.py

```diff
@@ -99,22 +118,22 @@ def fobj(x):
             return self.calcResiduals(x, ws_list)
 
         alatt0 = [a, b, c, alpha, beta, gamma]
-        try:
-            alatt, cov, info, msg, ier = leastsq(fobj, x0=alatt0, full_output=True)
-            # eval the fobj at optimal solution to set UB (leastsq iteration stops at a next sub-optimal solution)
-            fobj(alatt)
-        except ValueError:
-            logger.error("CalculateUMatrix failed - check initial lattice parameters and tolerance provided.")
-            return
-
+        alatt, cov, info, msg, ier = leastsq(fobj, x0=alatt0, full_output=True)
+        # eval the fobj at optimal solution to set UB (leastsq iteration stops at a next sub-optimal solution)
+        fobj(alatt)
         success = ier in [1, 2, 3, 4] and cov is not None  # cov is None when matrix is singular
         if success:
             # calculate errors
-            dof = sum([self.child_IndexPeaks(ws, RoundHKLs=True) for ws in ws_list]) - len(alatt0)
+            n_peaks = [self.exec_child_alg("IndexPeaks", PeaksWorkspace=ws, RoundHKLs=True, CommonUBForAll=True)[0] for ws in ws_list]
+            dof = sum(n_peaks) - len(alatt0)
             err = np.sqrt(abs(np.diag(cov)) * (info["fvec"] ** 2).sum() / dof)
-            for wsname in ws_list:
+            for iws, wsname in enumerate(ws_list):
                 ws = AnalysisDataService.retrieve(wsname)
                 ws.sample().getOrientedLattice().setError(*err)
+                if n_peaks[iws] < _MIN_NUM_INDEXED_PEAKS:
+                    logger.warning(
+                        f"Workspace: {wsname}, has only {n_peaks[iws]} indexed peaks, fewer than the desired {_MIN_NUM_INDEXED_PEAKS}"
```

Comment:
Can you make this clearee that this is the number of peaks indexed after global B matrix optimisation?